# CH EN 273 - Final Exam

In [1]:
import numpy as np
π = np.pi
from scipy.integrate import quad
from scipy.optimize import fsolve
from IPython.display import display, Markdown, Latex

In [2]:
P_atmos = {
    "sea_level": 1.00,
    "provo": 0.849,
    "denver": 0.820,
    "kansas_city": 0.968,
    "boise": 0.906,
    "phoenix": 0.961,
    "dallas": 0.985,
    "tuscon": 0.910
} #atm

### PROBLEM 2

> On a warm humid day (30°C and 50% relative humidity) an air conditioner is producing air at a rate of 0.5 m3/min at 20°C. If it was drier outside and the same amount of power was used, the flow rate of air____________.

**Answer: Would increase.**

Reasoning: Wet air has a higher heat capacity (by volume) than dry air, so the energy required to cool the same volume is less for dry air. Since the energy stays the same, the volumetric flow rate must increase.

The above assumes that we are talking about the volumetric flow rate. Mass flow rate, I would be less sure about, as the wet air would also be denser than the dry air.

### PROBLEM 3

>On a warm day (30°C) you’re drinking a cold beverage (5°C) in Atlanta, Georgia, where the relative humidity is 80%. Do you need a coaster? What if you are in Moab, Utah on a relatively cold evening (5°C) when the relative humidity is 60%?

**Answer: Yes, No.**

Downtown Atlanta has an elevation of 1050 ft above sea level, and an atmospheric pressure of about 0.96 atm at 30C.
Checking the Psychometric charts for 1.00 atm, I find the spot where the wet-bulb temp is 30C and humidity is 80%, then follow it horizontally (keeping mass constant) to find that it will hit 100% humidity (the dew point) when to cools to somewhere between 25 and 30 C. 5 C is quite a bit colder than 25 C, so it is reasonable to assume that there will be significant condensation from the glass.

For Moab, Utah, however, I would say no. Since the beverage is the same temperature as the ambient air, and the ambient air isn't at 100% relative humidity, there is no reason for there to be significant condensation.

### PROBLEM 4

> An endothermic reaction takes place in a continuous adiabatic reactor at constant pressure. There is one inlet and one outlet. The enthalpy of the outlet stream is:

**Answer: The same as the inlet.**

I am assuming the reactor is steady-state (IE, no accumulation)

Reasoning: Since the system is Adiabatic, no energy is flowing in or out of the system. As such, the total enthalpy coming in is going to be equal to the enthalpy going out, regardless of whether there is a reaction or whether that reaction is exothermic or endothermic. It's safe to assume, however, that the temperatures will not be the same.

### PROBLEM 5

> An electric fan is turned on in a well-insulated room with all the doors closed. The room temperature ___________.

**Answer: more information needed (increases OR stays the same).**

Reasoning: We've been told that the room is well-insulated, and the doors are closed. As such, I'd assume that air isn't leaving the room (closed system), and that the heat entering/leaving through the walls is negligible (temperature-wise adiabatic).

From an energy balance standpoint, we have work being done by the fan (to move the air around), drawing energy from the outlet (and that from outside the room). This means that the energy inside the room is increasing over time. Initially, that would be mostly velocity, but I'm sure the fan would heat up (no fan is 100% effective), and the rate of airflow would eventually level out (ie, the kinetic energy would stop increasing), which means that it's reasonable to imply that the air would eventually get warmer. As such, in a *perfectly sealed, adiabatic system* the temperature would slowly rise. This would lead me to say that the temperature **increases**.

However, since the system is *not perfectly* sealed/adiabatic (just well insulated, with closed doors), and the rate at which the room would heat up seems like it would be quite small, I don't know that it actually would increase. If the impact of the fan is in the same order of magnitude as the impact of air-flow under the door or of heat transfer through the walls (which have been stated as negligible), then you could assume that it would **stay the same**. It's hard to know without knowing the Wattage of the fan, the size of the room, and the actual flow of energy into the walls/air out from under the door. So I would go with **more information needed**

### PROBLEM 6

> A 10,000 gallon water tank atop a building delivers fresh drinking water for its occupants. A top floor resident and bottom floor resident are filling their bathtubs.  Assuming frictionless pipes and isothermal flow, which one will fill faster?

**Answer: The bottom floor resident.**

Reasoning: Energy is constant. The higher water will have higher potential energy. The lower water will consequently need to have higher thermal, chemical, or kinetic energy unless it does work on something on its way down. Thermal is ruled out (isothermal). Chemical is also ruled out (same water, nothing to cause one to react but not the other). And we've been told it's frictionless, which implies it's not doing work. That leaves just kinetic energy: IE the water at the bottom is flowing faster than the water at the top. 

### PROBLEM 7

> A sample of a solid compound begins at 65 °C. Heat is applied to increase the temperature by 46 °F. The sample is then cooled so that the temperature is reduced by -23 °R. What is the final temperature of the substance in units of K? (To be clear, the temperature first increases then decreases.)
>
> Report your answer with 2 decimal point accuracy. Is the problem is impossible, enter 99999.

In [3]:
T_init = 65 # C
ΔT_1__F = 46 # F - temperature differences are the same between F and R
ΔT_2__F = -23 # F - temperature differences are the same between F and R

ΔT_1__C = ΔT_1__F*5/9
ΔT_2__C = ΔT_2__F*5/9

T_final__C = T_init+ΔT_1__C+ΔT_2__C

T_final__K = T_final__C + 273.15
display(Markdown(rf"$T_\text{{final}} = {T_final__K:.2F} \text{{ K}}$"))

$T_\text{final} = 350.93 \text{ K}$

**Answer: 350.93 K**

Using the Conversion for C to K from Zybooks:
> T(K) = T(°C) + 273.15

### PROBLEM 8

> Pure ethanol (0.95 kg) is heated from 39 to 95 °C at a constant pressure of 1 atm. How much heat is required? Report your answer in units of kJ to 1 decimal point accuracy.

Ethanol is liquid at 39 C, and boils at around 78.5 C. As such, we will need to calculate the enthalpy change from 39 C to around 78.5 C (using liquid heat capacity) then the heat of vaporization, followed by the enthalpy change from around 78.5 C to 95 C (using gas heat capacity).

**Since I don't have another equation for liquid heat capacity than that in Zybooks, I'll be using it from 39 to 78 even though the equation says High T of 75.**

In [4]:
P = 760 # mmHg = 1 atm

# I initially couldn't find the boiling point of EtOH in Zybooks, so this would have calculated it from T_sat at 1 atm.
EtOH_PSAT_COEFS = {
    'A': 8.13484,
    'B': 1662.48,
    'C': 238.131
}
T_sat__EtOH = EtOH_PSAT_COEFS['B']/(EtOH_PSAT_COEFS['A'] - np.log10(P)) - EtOH_PSAT_COEFS['C']

# display(Markdown(rf"$T_\text{{sat}}$ of Ethanol at 1 atm = ${T_sat__EtOH:.2f} C$"))

T_1 = 39 # C (liquid)
T_v = 78.5 # C (boiling point)
T_2 = 95 # C (gas)

m = 0.95 * 1000 # g EtOH
MW_EtOH = 46.07 # g/mol from Zybooks

n = m / MW_EtOH # mol EtOH

In [5]:
Cp_COEFS_EtOH_liquid = {
    'a': 9.839E-02,
    'b': 0,
    'c': 0,
    'd': 0
} # liquid, 25 C to 75 C. kJ/(mol*C). Requires T in C.

Cp_COEFS_EtOH_gas = {
    'a': 6.134E-02,
    'b': 1.572E-04,
    'c': -8.749E-08,
    'd': 1.983E-11
} # gas, 0 C to 1200 C. kJ/(mol*C). Requires T in C.

def C_p(T, COEFS):
    return COEFS['a'] + COEFS['b']*T + COEFS['c']*T**2 + COEFS['d']*T**3

C_p(75, Cp_COEFS_EtOH_gas) # kJ/(mol*C) ### Used to test the equation. Compared to DIPPR, it's close enough I don't think I made any major mistakes.

0.07264623453125

In [6]:
Δh_1 = quad(lambda T: C_p(T,Cp_COEFS_EtOH_liquid),T_1,T_v)[0] # kJ/mol
Δh_v = 39.4 #kJ/mol from Zybooks
Δh_2 = quad(lambda T: C_p(T,Cp_COEFS_EtOH_gas),T_v,T_2)[0] # kJ/mol

ΔH = (Δh_1 + Δh_v + Δh_2) * n #kJ
display(Markdown(rf"$\Delta{{H}} = {ΔH:.1f} \text{{ kJ}}$"))

$\Delta{H} = 917.9 \text{ kJ}$

### PROBLEM 9

> Methanol may be produced by reacting carbon monoxide and hydrogen at moderate pressure. Consider a case where a stoichiometric feed of CO and H2 are fed in a single line to a reactor. The total molar flowrate of the inlet stream is 98.3 mol/min. The conversion is 100%, and the reactor has only one outlet stream. The inlet and outlet streams are at 25 °C and 125 °C, respectively. Determine the heat transfer rate. Report your answer in kW with 1 decimal place accuracy.
>
> The entire process occurs at a constant pressure of 5 atm, all compounds are in the gas phase, and all gases may be considered ideal.

Reaction:
$$
CO + 2H_2 \rightarrow H_3COH
$$

Using heat of formation + heat capacity to measure the energy in, then the energy out, to find the required energy.

In [7]:
P = 5 # atm

ṅ__in = 98.3 # mol/min

x_CO__in = 1/3
x_H2__in = 1 - x_CO__in

f_conv = 1.00
T__in = 25 # C
T__STD = 25 # C
T__out = 125 # C

In [8]:
# Heat of Formation is at 25C
Δhf_H2 = 0 #kJ/mol
Δhf_CO = -110.52 #kJ/mol
Δhf_MeOH = -201.2 #kJ/mol

Cp_COEFS_MeOH_gas = {
    'a': 4.293E-02,
    'b': 8.301E-05,
    'c': -1.870E-08,
    'd': -8.030E-12
} # gas, 0 C to 700 C. kJ/(mol*C). Requires T in C.
Δh_MeOH__out = quad(lambda T: C_p(T,Cp_COEFS_MeOH_gas),T__STD,T__out)[0] # kJ/mol from 25 to 125

In [9]:
ṅ_CO__in = x_CO__in * ṅ__in
ṅ_H2__in = ṅ__in - ṅ_CO__in

ξ = ṅ_CO__in * f_conv

ṅ_MeOH__out = ξ

# Since the inlet is at 25 C (which is also standard for the heat of formation) there is no need to add inlet heat capacity.
Q̇_in__toSTP = 0 # kJ/min ##relative to standard heat of formation
Q̇_react__STP = Δhf_MeOH * ṅ_MeOH__out - Δhf_CO * ṅ_CO__in - Δhf_H2 * ṅ_H2__in #kJ/min energy needed for the reaction at STP
Q̇_out__to125C = Δh_MeOH__out * ṅ_MeOH__out #kJ/min energy needed to heat the products to 770

Q̇_needed = (Q̇_out__to125C + Q̇_react__STP - Q̇_in__toSTP) / 60 #kJ/min * 1min/60sec => kW

display(Markdown(rf"$\dot{{Q}}_\text{{needed}} = {Q̇_needed:.1f} \text{{ kW}}$"))

$\dot{Q}_\text{needed} = -46.8 \text{ kW}$

In this case, the heat transfer is negative, in that the heat is produced by the reaction and needing to be transfered out.

### PROBLEM 10

> Use the following information for this problem. (The standard state for the heats of formation is 25 °C and 1 atm.)

| Component | Δℎ∘𝑓 (kJ/mol) | 𝐶𝑝 (kJ mol-1 K-1) |
| --- | --- | --- |
| A | -120 | 0.15 |
| B | -200 | 0.3 |
| C | -100 | 0.1 |
| D | -130 | 0.05 |

> What is the heat of reaction at 770 °C for the reaction below?
>
> 2 A + B → 1C + 4D
>
> Report your answer in units of kJ/mol with 1 decimal point accuracy.
>
>(All species are in the gas phase, and the reaction occurs at 1 atm.)

For this, I'm first going to calculate the change in energy from 770 C to 25 C for the reactants, then the difference in heat of formation, then the change in energy from 25 C to 770 C for the products.

In [10]:
Δhf_A = -120 #kJ/mol
Δhf_B = -200 #kJ/mol
Δhf_C = -100 #kJ/mol
Δhf_D = -130 #kJ/mol

Cp_A = 0.15 #kJ/(mol*K)
Cp_B = 0.3  #kJ/(mol*K)
Cp_C = 0.1  #kJ/(mol*K)
Cp_D = 0.05 #kJ/(mol*K)

n_A = 2 # A reacting per reaction
n_B = 1 # B reacting per reaction
n_C = 1 # C produced per reaction
n_D = 4 # D produced per reaction

T_react = 770 # C
T_STD = 25 # C

In [11]:
# heat of reactants relative to STD
Δh_reactants_770C = (n_A*Cp_A + n_B*Cp_B) * (T_STD - T_react) # energy required to change temperature

# heat of reaction at STD
Δh_reaction__STD = n_C*Δhf_C + n_D*Δhf_D - n_A*Δhf_A - n_B*Δhf_B # kJ/mol energy required for the reaction. If negative, then energy is released.

# heat of products relative to STD
Δh_products__770C = (n_C*Cp_C + n_D*Cp_D) * (T_react - T_STD) # kJ/mol energy required to change temperature

Δh_reaction = Δh_reactants_770C + Δh_reaction__STD + Δh_products__770C
display(Markdown(rf"$\Delta{{h}}_{{\text{{reaction}},770 C}} = {Δh_reaction:.1f} \frac{{\text{{kJ}}}}{{\text{{mol}}}}$"))

$\Delta{h}_{\text{reaction},770 C} = -403.5 \frac{\text{kJ}}{\text{mol}}$

### PROBLEM 11

>A large tropical greenhouse is being built in a city in Colorado with an elevation of 1500 m above sea level. The environment inside the greenhouse must approximate that of the native environment of the plants and insects being housed. In the winter, this requires heating and humidification as the natural conditions in this area of Colorado during this time of year is cold and dry.
>
>A stream (T = 16 °C), consisting of 200 kg/min dry air and 2.5 lb/min water, is heated at a rate of 50 kW. What is the final relative humidity (RH) of the stream after heating?

In [12]:
T_1 = 16 # C
ṁ_air = 200 / 60 # kg/s dry air
ṁ_water = 2.5 / 2.20462 / 60 # kg/s water from lb/min
Q̇ = 50 # kW

In [13]:
humidity_ratio = ṁ_water/ṁ_air
humidity_ratio

0.005669911367945497

In [14]:
h_in = 31 #kJ/kg Dry Air from Psychrometric charts at 1500 ft elevation. Intersection of 16 C (dry bulb) and humidity ratio of 0.0055
Ḣ_in = h_in * ṁ_air # kJ/s = kW
Ḣ_out = Ḣ_in + Q̇ # kW
h_out = Ḣ_out / ṁ_air  #kJ/kg Dry Air
h_out

46.0

From the psychrometric charts at 1500 ft elevation, the intersection of humidity ratio of 0.0055 (which should be constant) and specific enthalpy of 46 kJ/kg Dry Air is at around **18% RH** and 30 C.

### PROBLEM 12

> The American Petroleum Institute (API) is a national trade association for oil and gas processing. They define standard conditions as 60 °F and 1 atm.
> 
> Pentane at 136 °F and 22.4 psia flows at a rate of 1.07 ft3/min. What is the flowrate in SCIM (standard cubic inches per minute) for the API standard?

In [15]:
R_imp = 10.73 # psia·ft3/lbmol·°R
T_STD__API = 60 + 459.67 # °R
P_STD__API = 14.696 # psia
v_STD__API = R_imp * T_STD__API / P_STD__API # ft3/lbmol
T = 136 + 459.67 # °R
P = 22.4 # psia
V̇ = 1.07 # ft3/min
ṅ = P * V̇ / (R_imp * T) # lbmol/min

In [16]:
V̇_STD__API = ṅ * v_STD__API * 12**3 # in3/min or SCIM
display(Markdown(rf"$\dot{{V}}_\text{{STD,API}} = {V̇_STD__API:.0f} \text{{ SCIM}}$"))

$\dot{V}_\text{STD,API} = 2459 \text{ SCIM}$

### PROBLEM 13

> Methanol is a chemical with a variety of uses such as inks, resins, adhesives, fuel. It is also used as a building block to synthesize many other chemicals. One way to make methanol synthesis more sustainable is to generate syngas (a combination of CO and H2) from combustion of biomass and then make methanol from the syngas through the following reaction.
>
> CO + 2 H2 = CH3OH
>
> Consider the reactor below. The feed is a mixture of syngas at 207 °C. Determine the temperature of the exiting stream if the reactor operates adiabatically. (All species are in the gas phase and behave ideally.)
> ![PFD](https://learningsuite.byu.edu/plugins/Upload/fileDownload.php?fileId=fl0ccIidiUCq&service=scout&appId=student&courseID=NWKvdiPaysDR)
> 
> Report your answer in the box below, and upload your work to the "Final Exam Partial Credit" assignment on Learning Suite.

Start from energy balance

$$
\frac{d}{dt}\left(mu+PE+KE\right)_{\text{mis}} = -\sum_i^{N_{\text{streams}}}\nu\dot{m_i}\left(h_i+\frac{1}{2}s_i^2+gz_i\right) + \dot{Q} + \dot{W}_{\text{nonstream }Pv}
$$

Simplify (Steady State, No Work, Adiabatic (except for reaction))

$$
0 = -\sum_i^{N_{\text{streams}}}\nu\dot{m_i}\left(h_i\right) + \dot{Q}_\text{react}
$$

Which Becomes

$$
\sum_\text{inlets}^{N_{\text{streams}}}\dot{m_i}h_i + \dot{Q}_\text{react} = \sum_\text{exits}^{N_{\text{streams}}}\dot{m_e}h_e
$$

Asking for $T_\text{out}$



Next steps would be to get an energy path from 207 C to standard (25 C), then across the reaction. That will tell us how much energy needs to leave with the products. We can then setup a function to calculate the enthalpy of products based on their temperature. Setting them equal to each other (overall adiabatic) in fsolve will get us the temperature we are looking for.

In [17]:
T__in = 207 # C
T__STD = 25 # C

ṅ_CO__in = 22 # mol/s
ṅ_H2__in = 88 # mol/s

ṅ_CO__out = 2.2 # mol/s
ṅ_H2__out = 48.4 # mol/s
ṅ_MeOH__out = 19.8 # mol/s

# Used to test that the flow rates worked stoichiometrically. They did.
ξ = ṅ_MeOH__out #mol/s

In [18]:
# Heat of Formation is at 25C
Δhf_H2 = 0 #kJ/mol
Δhf_CO = -110.52 #kJ/mol
Δhf_MeOH = -201.2 #kJ/mol

Cp_COEFS_H2_gas = {
    'a': 2.884E-02,
    'b': 7.650E-08,
    'c': 3.288E-09,
    'd': -8.698E-13
} # gas, 0 C to 1500 C. kJ/(mol*C). Requires T in C.

Cp_COEFS_CO_gas = {
    'a': 2.895E-02,
    'b': 4.110E-06,
    'c': 3.548E-09,
    'd': -2.220E-12
} # gas, 0 C to 1501 C. kJ/(mol*C). Requires T in C.

Cp_COEFS_MeOH_gas = {
    'a': 4.293E-02,
    'b': 8.301E-05,
    'c': -1.870E-08,
    'd': -8.030E-12
} # gas, 0 C to 700 C. kJ/(mol*C). Requires T in C.

In [19]:
Δh_H2__in = quad(lambda T: C_p(T,Cp_COEFS_H2_gas),T__in,T__STD)[0] # kJ/mol from 207 to 25 C. Negative, because heat is not needed, but released.
Δh_CO__in = quad(lambda T: C_p(T,Cp_COEFS_CO_gas),T__in,T__STD)[0] # kJ/mol from 207 to 25 C. Negative, because heat is not needed, but released.

Q̇_reactants__toSTD = Δh_H2__in * ṅ_H2__in + Δh_CO__in * ṅ_CO__in # kJ/s = kW. this will be negative, since energy is released, not generated.
Q̇_reaction__STD = ξ * Δhf_MeOH - ξ * Δhf_CO - 2 * ξ * Δhf_H2 # kW # negative since energy is released, not needed

def Q̇_products__toT(T):
    Δh_H2__out = quad(lambda T: C_p(T,Cp_COEFS_H2_gas),T__STD,T)[0] # kJ/mol, positive since energy is needed
    Δh_CO__out = quad(lambda T: C_p(T,Cp_COEFS_CO_gas),T__STD,T)[0] # kJ/mol, positive since energy is needed
    Δh_MeOH__out = quad(lambda T: C_p(T,Cp_COEFS_MeOH_gas),T__STD,T)[0] # kJ/mol, positive since energy is needed
    return Δh_H2__out * ṅ_H2__out + Δh_CO__out * ṅ_CO__out + Δh_MeOH__out * ṅ_MeOH__out # kW

def Q̇(T): # Q̇ [kW], T [C], overall energy balance. Should be 0, since it's overall adiabatic
    return Q̇_reactants__toSTD+Q̇_reaction__STD+Q̇_products__toT(T)

#test the function
Q̇(200)

-1940.2003483199169

In [20]:
T__out = fsolve(Q̇,600)[0]
display(Markdown(rf"$T_\text{{out}} = {T__out:.1f} C$"))

$T_\text{out} = 833.0 C$

### PROBLEM 14

> Many acids are used in the semiconductor industry to make computer chips. You are performing a safety review of the cylindrical tanks (dimensions below) that will be used to store such acids at a new semiconductor fabrication plant (fab) being built. Your task is to model how quickly acid (rho = 1.05 g/mL) will escape if a leak develops in the tank.
>
> ![Tank Diagram](https://learningsuite.byu.edu/plugins/Upload/fileDownload.php?fileId=d9NtUMSk_ABv&service=scout&appId=student&courseID=NWKvdiPaysDR)
> 
> Consider a case where a hole develops at the bottom of an initially-full tank. The size of the hole increases with time as the flow erodes the tank material. The volumetric flowrate of the leak (𝑉˙, L/min) as a function of time (𝑡, min) is given by
> $\dot{V} \left[\frac{\text{L}}{\text{min}}\right]= 2 + \frac {4} {\text{min}} t$
>
> How long (in minutes) does it take for half the contents of the tank to drain?
>
> Report your answer in the box below, and upload your work to the "Final Exam Partial Credit" assignment on Learning Suite.

I'm starting by calculating the volume of liquid in the tank (V_0) at the start, then writing up a function for the leakage at a point in time. Another function allows us to use an integral to subtract the amount leaked from the original amount to give us the amount left in the tank at any point in time.

We can then use fsolve to see at what point in time the tank's content will be half the original amount.

As it turns out, the density is irrelevant to this problem.

In [21]:
ρ = 1.05 # g/mL
D = 1.5 # m
z = 2.5 # m

# full volume
V_0 = π*D**2/4 * z * 1000 # L from m3 (πhD^2/4)

def V̇_leak(t): # V̇ [L/min], t [min]
    return 2 + 4 * t

def V_remaining(t): # V [L], t [min]
    return V_0 - quad(V̇_leak,0,t)[0]

# test the function before passing it to fsolve
V_remaining(20)

3577.864669110647

In [22]:
# solve for the point in time where the amount left is exactly half of the original amount
t_half = fsolve(lambda t: (V_0/2.0 - V_remaining(t)), 20)[0]
display(Markdown(rf"$t_\text{{half}} = {t_half:.2f} \text{{ min}}$"))

$t_\text{half} = 32.74 \text{ min}$